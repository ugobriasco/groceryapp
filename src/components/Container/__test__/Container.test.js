import React from 'react';
import { View } from 'react-native';
import renderer from 'react-test-renderer';
//import styles from '../styles';
import { Container, styles} from '../index.js';

// Note: test renderer must be required after react-native.


it('renders correctly', () => {
  const rendered = renderer.create(<Container/>).toJSON();
  expect(rendered).toBeTruthy();
});

it('styles is an object', () => {
expect(typeof styles).toBe('object');
});


it('renders the child prop', () => {
	const rendered = renderer.create(<Container><View /></Container>).toJSON();
	expect(rendered).toMatchSnapshot();
});

it('uses the specified background, if provided', () => {
	const rendered = renderer.create(<Container backgroundColor="red"/>).toJSON();
	expect(rendered).toMatchSnapshot();
});