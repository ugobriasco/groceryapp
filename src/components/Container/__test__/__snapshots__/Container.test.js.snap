// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the child prop 1`] = `
<View
  style={
    Array [
      undefined,
    ]
  }
>
  <View />
</View>
`;

exports[`uses the specified background, if provided 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "backgroundColor": "red",
      },
    ]
  }
/>
`;
